### PR TITLE
Added -L option

### PR DESCRIPTION
**Pull request by RedocamaI**
Added the '-L' option, which will be used for generating the desired length of the random password.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,6 +1,22 @@
...
+    if(argc > 4)
+    {
+        // tell the user how to run the program
+        std::cout<<"Usage1: passgen -L length_of_the_password_you_want\n";
+        // We can add other usage messages here based on the input error committed by the user.
+        return 1;
+    }
+
+    // set the length of the character set
+    if(argv[1] == "-L")
```
Comment (anjanik012) [inline src/main.cpp]: you must convert `argv[1]` to `std::string` as it has overloaded operator `==` for string comparision. If you don't want to convert it to the string object and then compare, you can use C `strcmp` function here which is also faster than converting to `std::string` object.

File: src/main.cpp [hunk 2]
```diff
@@ -1,6 +1,22 @@
...
+    {
+        int length = atoi(argv[2]);
+        // generate length using length generator.
```
Comment (anjanik012) [inline src/main.cpp]: There are a couple of problems with this approach to parse. Your method assumes that `-L` option is present in `argv[1]` which may not always be the case as we also have a `-S` option which may come before `-L`. This needs rewrite

Comment (RedocamaI) [reply src/main.cpp]: Yes, just changed the code. And I think, wouldn't it be feasible to design a specific format for command-line arguments, for example," passgen -S -L 20". That is we can add this instruction in how to use the generator in the "-H' section.

File: src/main.cpp [hunk 3]
```diff
@@ -11,11 +12,19 @@ int main(int argc, char *argv[]){
...
+    for(int i = 0; i<argc; i++)
     {
-        int length = atoi(argv[2]);
-        // generate length using length generator.
-
+        if(strcmp(argv[i], "-L") == 0)
+        {
+            // And since the length entered by the user may be after '-S' so we
+            // have to take another loop to identify where it is.
+            for(int j = i; j<argc; j++)
```
Comment (anjanik012) [inline src/main.cpp]: There is no need for this loop. If -L is in position i then value will be in position i+1

**Review by anjanik012 — COMMENTED** [review]
Reformat the code. No braces on new lines.

File: src/main.cpp [hunk 4]
```diff
@@ -1,6 +1,27 @@
 #include <iostream>
+#include <string.h>
 
 int main(int argc, char *argv[]){
-    std::cout << "Hello,World!" << std::endl;
+    // set a limit of the number of arguments to be passed: in argc
+    if(argc > 4)
+    {
```
Comment (anjanik012) [inline src/main.cpp]: opening brace on new line

File: src/main.cpp [hunk 5]
```diff
@@ -1,6 +1,27 @@
...
+        // tell the user how to run the program
+        std::cout<<"Usage1: passgen -L length_of_the_password_you_want\n";
+        // We can add other usage messages here based on the input error committed by the user.
+        return 1;
+    }
+
+    // set the length of the character set to default, which is 8.
+    int length = 8;
+    for(int i = 0; i<argc; i++)
+    {
```
Comment (anjanik012) [inline src/main.cpp]: opening brace on new line

File: src/main.cpp [hunk 6]
```diff
@@ -1,6 +1,27 @@
...
+        if(strcmp(argv[i], "-L") == 0)
```
Comment (anjanik012) [inline src/main.cpp]: add braces for if

File: src/main.cpp [hunk 7]
```diff
@@ -1,6 +1,27 @@
...
+            length = atoi(argv[i+1]);
+    }
+    
     return 0;
 }
+
```
Comment (anjanik012) [inline src/main.cpp]: what is this?

**Review by anjanik012 — COMMENTED** [review]
As I said, opening braces shouldn't be in new lines

**Pull request merged**